src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -233,16 +239,96 @@ public void doRun() {
 
     private void upsertJob(final BuildConfig buildConfig) throws Exception {
         if (isPipelineStrategyBuildConfig(buildConfig)) {
+            // check if Multibranch annotation is present
+            Boolean enableMultibranchSync = Boolean.valueOf(getAnnotation(buildConfig, ENABLE_MULTIBRANCH_SYNC));
             // sync on intern of name should guarantee sync on same actual obj
             synchronized (buildConfig.getMetadata().getUid().intern()) {
                 ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
                     @Override
                     public Void call() throws Exception {
-                        String jobName = jenkinsJobName(buildConfig);
-                        String jobFullName = jenkinsJobFullName(buildConfig);
-                        WorkflowJob job = getJobFromBuildConfig(buildConfig);
                         Jenkins activeInstance = Jenkins.getActiveInstance();
                         ItemGroup parent = activeInstance;
+                        String jobName = enableMultibranchSync ? jenkinsJobName(buildConfig) + " Multibranch" : jenkinsJobName(buildConfig);
+                        String jobFullName = enableMultibranchSync ? jenkinsJobFullName(buildConfig) + " Multibranch" : jenkinsJobFullName(buildConfig);
+
+                        if (enableMultibranchSync){
+                          parent = getFullNameParent(activeInstance, jobFullName, getNamespace(buildConfig));
+                          System.out.println("_________________ Multibranch sync enabled _____________________");
+                          WorkflowMultiBranchProject multibranchProject = new WorkflowMultiBranchProject(parent, jobName);
```

Comment:
To your question, about setting the FlowDefinition, I think you 
- extend/subclass WorkflowMultiBranchProject 
- override the org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject.newProjectFactory() method in your new class, and in turn create an extension/subclass of org.jenkinsci.plugins.workflow.multibranch.WorkflowBranchProjectFactory
- In that new class, you've got a org.jenkinsci.plugins.workflow.multibranch.WorkflowBranchProjectFactory.createDefinition() method you can override
- override that method to do what we do in io.fabric8.jenkins.openshiftsync.BuildConfigToJobMapper.mapBuildConfigToFlow(BuildConfig) except  instead of using org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition.CpsFlowDefinition(String, boolean) you are going to pass our jenkinsfile script into org.jenkinsci.plugins.workflow.multibranch.SCMBinder.SCMBinder(String)

Reply:
Okay, will work on that.